geocity/apps/api/serializers.py

```diff
@@ -42,150 +43,181 @@ def get_form_fields(
                     else ""
                 )
 
-                # List of a list, to split wot in objects. Check if last wot changed or never assigned. Means it's first iteration
-                if property and wot != last_wot:
-                    # Don't add values if there's only the "WOT" without any field
+                # List of a list, to split form in objects. Check if last form changed or never assigned. Means it's first iteration
+                if property and form != last_field:
+                    # Don't add values if there's only the "form" without any field
                     if len(property) > 1:
-                        wot_properties.append(property)
+                        fields_dict.append(property)
 
                     property = []
-                    # WOT
+                    # form
                     property.append(
                         {
                             "key": key_for_form,
-                            "value": wot,
+                            "value": form,
                             "type": "text",
                         }
                     )
 
-                if not last_wot:
+                if not last_field:
                     property.append(
                         {
                             "key": key_for_form,
-                            "value": wot,
+                            "value": form,
                             "type": "text",
                         }
                     )
 
-                last_wot = wot
+                last_field = form
 
                 # Show fields_values only when the current submission__administrative_entity
                 # is one of the administrative_entities associated to the user
                 # or user is the submission__author
                 # or show field_values that are designed as public in a public permit_request
-                if prop["field_values__field__input_type"] == "file" and (
+                if field["field_values__field__input_type"] == "file" and (
                     (
                         administrative_entities_associated_to_user_list
-                        and prop["submission__administrative_entity"]
+                        and field["submission__administrative_entity"]
                         in administrative_entities_associated_to_user_list
                     )
-                    or (current_user and prop["submission__author"] == current_user.id)
-                    or prop[
+                    or (current_user and field["submission__author"] == current_user.id)
+                    or field[
                         "field_values__field__is_public_when_permitrequest_is_public"
                     ]
                 ):
                     # get_property_value return None if file does not exist
-                    file = get_field_value_based_on_field(prop)
+                    file = get_field_value_based_on_field(field)
```

Comment:
Out of scope, but this might be a performance leak as it calls DB on each loop:
```
def get_field_value_based_on_field(prop):
    property_object = FieldValue.objects.get(
        field_id=prop["field_values__field_id"],
        selected_form_id=prop["id"],
    )

    return property_object.get_value()
```

We should make on call on DB for all Field Values and then manipulate them, ideally using vetorized operations, as we now have 🐼 onboard.

Reply:
Only used when it's a file